unordered_associative_task_2/student/src/impl.cpp

```diff
@@ -41,9 +72,55 @@ filtering_map groupDuplicates (const std::vector<std::string>& dataSource, Filte
     return grouped;
 }
 
-std::vector<std::vector<std::string> > findDuplicates(const std::string &rootPath)
+/** @todo With help of std filesystem, lists all regular files under specified directory recursively*/
+std::vector<std::string> listFiles(const std::string& directory)
+{
+    std::vector<std::string> fileNames;
+    for (auto& p : fs::recursive_directory_iterator(directory)) {
```

Comment:
Simple and readable - nice :)
You can try to implement it in STL way by combining algorithms, iterators and functors.
Think what your loop does: copies value of iterator p to fileNames with condition (predicate).